packages/server/integrations/BaseTaskIntegrationManager.ts

```diff
@@ -0,0 +1,54 @@
+import {GraphQLResolveInfo} from 'graphql'
+import {GQLContext} from '../graphql/graphql'
+import {getUserId} from '../utils/authorization'
+import Task from '../database/types/Task'
+import Team from '../database/types/Team'
+import {AtlassianAuth} from '../postgres/queries/getAtlassianAuthByUserIdTeamId'
+import {GitHubAuth} from '../postgres/queries/getGitHubAuthByUserIdTeamId'
+
+type Auth = AtlassianAuth | GitHubAuth
+
+export default abstract class BaseTaskIntegrationManager {
+  readonly context: GQLContext
+  readonly info: GraphQLResolveInfo
+  readonly teamId: string
+  readonly viewerId: string
+  readonly userId: string | null
+  readonly task: Task
+  readonly team: Team
+  readonly rawContentStr: string
+  readonly createdBySomeoneElse: boolean
```

Comment:
-2 Before this change the logic "If the task is pushed by someone else than the author, then add a comment" is now spread over 3 files, while before it was all in the mutation itself. I think this makes it much harder in the long run to maintain. Also the state which is stored in the base class versus the arguments in subclasses seem arbitrary and it seems conceptually wrong to have an instance depending on the `GQLContext` and `GraphQLResolveInfo`.

I think we would have a cleaner solution when we decouple this a bit and instantiate the `TaskIntegrationManager` with auth and userId

```
class TaskIntegrationManagerFactory {
  authLoaderKeys: Record<IntegrationProviderServiceEnumType, Loaders>
  getManager(service: IntegrationProviderServiceEnumType, userId, auth): TaskIntegrationManager

}

interface TaskIntegrationManager {
  createTask(rawContentStr: string, projectId: string): {
    integrationHash: string,
    integration: {
      accessUserId: string,
      service: IntegrationProviderServiceEnumType
    }
  }
  addCreatedBySomeoneElseComment(projectId, viewerName, assigneeName, teamName, teamDashboardUrl)
}
```

Reply:
@Dschoordsch What if we don't use the base class state at all? I made some changes based on your comments. Can you please check?